internal/services/apimanagement/api_management_api_tag_description_resource.go

```diff
@@ -0,0 +1,166 @@
+package apimanagement
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/apimanagement/mgmt/2021-08-01/apimanagement"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/apimanagement/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/apimanagement/schemaz"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/apimanagement/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func resourceApiManagementApiTagDescription() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceApiManagementApiTagDescriptionCreateUpdate,
+		Read:   resourceApiManagementApiTagDescriptionRead,
+		Update: resourceApiManagementApiTagDescriptionCreateUpdate,
+		Delete: resourceApiManagementApiTagDescriptionDelete,
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.ApiTagDescriptionsID(id)
+			return err
+		}),
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+
+			"tag_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.TagID,
+			},
+
+			"api_name": schemaz.SchemaApiManagementApiName(),
+
+			"description": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+			},
+
+			"external_documentation_url": {
+				Type:         pluginsdk.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.IsURLWithHTTPorHTTPS,
+			},
+
+			"external_documentation_description": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceApiManagementApiTagDescriptionCreateUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).ApiManagement.ApiTagDescriptionClient
+
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	tagId, err := parse.TagID(d.Get("tag_id").(string))
+	if err != nil {
+		return fmt.Errorf("parsing `api_id`: %v", err)
+	}
+
+	id := parse.NewApiTagDescriptionsID(tagId.SubscriptionId, tagId.ResourceGroup, tagId.ServiceName, d.Get("api_name").(string), tagId.Name)
```

Comment:
this seems weird...

```
//go:generate go run ../../tools/generator-resource-id/main.go -path=./ -name=Tag -id=/subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/resGroup1/providers/Microsoft.ApiManagement/service/service1/tags/tag1
//go:generate go run ../../tools/generator-resource-id/main.go -path=./ -name=ApiTag -id=/subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/resGroup1/providers/Microsoft.ApiManagement/service/service1/apis/api1/tags/tag1
```
this should be ApiTag?

Reply:
The ID format here is correct, this is for API tag(https://docs.microsoft.com/en-us/rest/api/apimanagement/current-ga/tag/assign-to-api?tabs=HTTP), 
This PR is to cover the API tag description, they are different resources(https://docs.microsoft.com/en-us/rest/api/apimanagement/current-ga/api-tag-description)

Reply:
I see,  I will try `azurerm_api_management_api_tag` instead of `azurerm_api_management_tag`